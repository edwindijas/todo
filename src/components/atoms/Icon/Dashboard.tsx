import React from 'react'
import ReactComponent from '@material-design-icons/svg/round/dashboard.svg?react'
import Icon from '.'

const DashboardIcon = (props: React.SVGAttributes<SVGElement>): JSX.Element => (
  <Icon {...props} component={ReactComponent} />
)

export default DashboardIcon
