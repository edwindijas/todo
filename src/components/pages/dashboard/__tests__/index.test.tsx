import '@testing-library/jest-dom/jest-globals';
import '@testing-library/jest-dom';
import { waitFor } from '@testing-library/react'
import Dashboard from '..'
import render from '@/utils/test-utils/customRender'

it ('Renders snapshot correctly', async () => {
  const {container, getByTestId} = render(<Dashboard />)
  await waitFor(() => { expect(getByTestId('page-dashboard')).toBeInTheDocument() })
  expect(container).toMatchSnapshot()
})
