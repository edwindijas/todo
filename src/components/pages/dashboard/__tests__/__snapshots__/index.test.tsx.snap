// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders snapshot correctly 1`] = `
<div>
  <div
    class="h-full flex bg-bg"
  >
    <header>
      <div
        class="w-64 h-full bg-white relative"
      >
        <div>
          <div
            class="flex py-4 px-6 gap-2 items-center"
          >
            <div
              class="w-8 h-8"
            >
              <svg
                class=" fill-blue-dark w-full h-full"
                data-testid="component-atom-icon"
              />
            </div>
            <p
              class="text-md text-blue-dark flex flex-col gap-0 leading-4"
            >
              <span>
                TASK
              </span>
              <span
                class="font-extrabold"
              >
                SPACE
              </span>
            </p>
          </div>
          <div
            class="px-4 pt-4"
          >
            <label
              class="  flex relative cursor-text border-[#A6B7D0] bg-[#F4F8FC]  fill-blue-desaturated  border rounded-full h-12 px-2"
            >
              <div
                class="w-7 shrink-0"
              >
                <svg
                  class=" w-full h-full"
                  data-testid="component-atom-icon"
                />
              </div>
              <div
                class="relative"
              >
                <span
                  class="absolute flex top-0 bottom-0 left-0 items-center text-blue-desaturated opacity-100"
                >
                  Search
                </span>
                <input
                  class="block outline-none w-full h-full bg-transparent h-12"
                  value=""
                />
              </div>
            </label>
          </div>
          <div
            class="p-4"
          >
            <ul
              class="flex flex-col"
            >
              <li>
                <a
                  aria-current="page"
                  class=" flex capitalize items-center gap-2 relative px-4 h-12 rounded-md bg-blue-light text-blue-medium fill-blue-medium relative before:absolute before:top-2 before:bottom-2 before:left-0 before:bg-blue-medium before:w-1 before:block before:rounded-sm"
                  href="/"
                >
                  <div
                    class="shrink-0 h-6 w-6"
                  >
                    <svg
                      class=" w-full h-full"
                      data-testid="component-atom-icon"
                    />
                  </div>
                  <span
                    class="w-full"
                  >
                    dashboard
                  </span>
                </a>
              </li>
            </ul>
          </div>
          <div
            class="px-6 font-bold flex flex-col gap-3 border-t-2 border-b py-3 border-bg"
          >
            <p
              class="text-sm text-blue-medium"
            >
              Projects
            </p>
            <div
              class="flex justify-between items-center"
            >
              <p
                class="text-blue-dark"
              >
                Office Work
              </p>
              <div>
                <svg
                  class="w-5 h-5 fill-blue-dark w-full h-full"
                  data-testid="component-atom-icon"
                />
              </div>
            </div>
          </div>
          <div
            class="p-4"
          >
            <ul
              class="flex flex-col"
            >
              <li>
                <a
                  class=" flex capitalize items-center gap-2 relative px-4 h-12 rounded-md text-blue-desaturated fill-blue-desaturated"
                  href="/tasks"
                >
                  <div
                    class="shrink-0 h-6 w-6"
                  >
                    <svg
                      class=" w-full h-full"
                      data-testid="component-atom-icon"
                    />
                  </div>
                  <span
                    class="w-full"
                  >
                    tasks
                  </span>
                </a>
              </li>
              <li>
                <a
                  class=" flex capitalize items-center gap-2 relative px-4 h-12 rounded-md text-blue-desaturated fill-blue-desaturated"
                  href="/calendar"
                >
                  <div
                    class="shrink-0 h-6 w-6"
                  >
                    <svg
                      class=" w-full h-full"
                      data-testid="component-atom-icon"
                    />
                  </div>
                  <span
                    class="w-full"
                  >
                    calendar
                  </span>
                </a>
              </li>
              <li>
                <a
                  class=" flex capitalize items-center gap-2 relative px-4 h-12 rounded-md text-blue-desaturated fill-blue-desaturated"
                  href="/board"
                >
                  <div
                    class="shrink-0 h-6 w-6"
                  >
                    <svg
                      class=" w-full h-full"
                      data-testid="component-atom-icon"
                    />
                  </div>
                  <span
                    class="w-full"
                  >
                    board
                  </span>
                </a>
              </li>
              <li>
                <a
                  class=" flex capitalize items-center gap-2 relative px-4 h-12 rounded-md text-blue-desaturated fill-blue-desaturated"
                  href="/project-settings"
                >
                  <div
                    class="shrink-0 h-6 w-6"
                  >
                    <svg
                      class=" w-full h-full"
                      data-testid="component-atom-icon"
                    />
                  </div>
                  <span
                    class="w-full"
                  >
                    project-settings
                  </span>
                </a>
              </li>
              <li>
                <a
                  class=" flex capitalize items-center gap-2 relative px-4 h-12 rounded-md text-blue-desaturated fill-blue-desaturated"
                  href="/members"
                >
                  <div
                    class="shrink-0 h-6 w-6"
                  >
                    <svg
                      class=" w-full h-full"
                      data-testid="component-atom-icon"
                    />
                  </div>
                  <span
                    class="w-full"
                  >
                    members
                  </span>
                </a>
              </li>
            </ul>
          </div>
        </div>
        <div
          class="px-6 absolute bottom-4 w-full"
        >
          <a
            class="w-full justify-center flex capitalize items-center gap-2 relative text-center border border-blue-medium bg-blue-light text-blue-medium fill-blue-medium px-2 h-12 rounded-md"
            href="/"
          >
            <div
              class="shrink-0 h-6 w-6 absolute top-0 bottom-0 my-auto left-4"
            >
              <svg
                class=" w-full h-full"
                data-testid="component-atom-icon"
              />
            </div>
            <span
              class="w-full"
            >
              New Tasks
            </span>
          </a>
        </div>
      </div>
    </header>
    <div
      class="w-full h-full"
    >
      <div>
        <div
          data-testid="page-dashboard"
        >
          Dashboard Coming Soon
        </div>
      </div>
    </div>
  </div>
</div>
`;
